prov/gni/src/gnix_ep.c

```diff
@@ -1167,21 +1162,17 @@ static inline struct gnix_fab_req *__find_tx_req(
 	GNIX_DEBUG(FI_LOG_EP_CTRL, "searching VCs for the correct context to"
 			" cancel, context=%p", context);
 
-	fastlock_acquire(&ep->vc_list_lock);
-	dlist_for_each(&ep->wc_vc_list, vc, entry)
-	{
-		GNIX_DEBUG(FI_LOG_EP_CTRL, "checking vc=%p\n", vc);
+	while ((vc = _gnix_nic_next_pending_vc(ep->nic))) {
```

Comment:
This will only look through "scheduled" VCs (having work to be done).  I don't think that means all VCs.  I think I need to revisit the way VCs are queued with TXs.  This probably works for now.